cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultSchedulerService.java

```diff
@@ -62,13 +63,14 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
       Trigger trigger = context.getTrigger();
       String key = trigger.getKey().getName();
       String[] parts = key.split(":");
-      Preconditions.checkArgument(parts.length == 5);
+      Preconditions.checkArgument(parts.length == 6);
 
       String namespaceId = parts[0];
       String applicationId = parts[1];
-      ProgramType programType = ProgramType.valueOf(parts[2]);
-      String programId = parts[3];
-      String scheduleName = parts[4];
+      String appVersion = parts[3];
```

Comment:
but what about existing schedules that have been persisted that only have 5 parts. The precondition on line 66 will fail right?